microprofile/fault-tolerance-smallrye/cdi/src/main/java/org/wildfly/extension/microprofile/faulttolerance/cdi/RequestContextCommandListener.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.microprofile.faulttolerance.cdi;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.jboss.weld.context.RequestContext;
+import org.jboss.weld.context.unbound.Unbound;
+import org.jboss.weld.manager.BeanManagerImpl;
+
+import io.smallrye.faulttolerance.CommandListener;
+import io.smallrye.faulttolerance.config.FaultToleranceOperation;
+
+/**
+ * Activates CDI request context when a Hystrix command that wraps a Fault Tolerance operation is executed.
+ *
+ * @author Martin Kouba
+ * @author Radoslav Husar
+ */
+@ApplicationScoped
+public class RequestContextCommandListener implements CommandListener {
+
+    private final BeanManagerImpl beanManager;
```

Comment:
Why inject the BeanManager implementation class instead of org.jboss.weld.manager.api.WeldManager?  Then you can remove the dependencies on the org.jboss.weld.core and org.jboss.weld.spi modules.

Reply:
That's really question for Martin, but doing what you suggested yields

Caused by: java.lang.ClassNotFoundException: org.jboss.weld.manager.api.WeldManager from [Module \"org.wildfly.microprofile.fault-tolerance-smallrye.cdi\" version 19.0.0.Beta1-SNAPSHOT from local module loader @6537cf78 (finder: local module finder @67b6d4ae (roots: /Users/rhusar/git/wildfly/testsuite/integration/microprofile/target/wildfly/modules,/Users/rhusar/git/wildfly/testsuite/integration/microprofile/target/wildfly/modules/system/layers/base))]

Reply:
The `org.jboss.weld.manager.api.WeldManager` is in the weld-spi module....

Reply:
Anyway, I dont see how can those be removed.

Reply:
You can at least remove the dependency on the org.jboss.weld.core module, no?

Reply:
No – that fails with CNF as well.